Not treating `@staticmethod`s as static within an `abc.ABCMeta` (sub)class

<!--
Thank you for taking the time to report an issue! We're glad to have you involved with Ruff.

If you're filing a bug report, please consider including the following information:

* A minimal code snippet that reproduces the bug.
* The command you invoked (e.g., `ruff /path/to/file.py --fix`), ideally including the `--isolated` flag.
* The current Ruff settings (any relevant sections from your `pyproject.toml`).
* The current Ruff version (`ruff --version`).
-->

While running `ruff 0.0.243` on some code, I had a confusing linting output, which can be reproduced with the following snippet:

```python
from abc import ABCMeta


class MetaClass(ABCMeta):
    @staticmethod
    def i_dont_get_cls_param(not_cls) -> bool:
        return False
```

Despite the method being noted as static, `ruff` is complaining that it should take a `cls` parameter:
```
$ ruff ruff.py
ruff.py:6:30: N804 First argument of a class method should be named `cls`
Found 1 error.
```

Based on the usage of this function in other code, it is indeed static, so this seems like it's some sort of precedence issue between treating all methods of `ABCMeta` as classmethods and the `@staticmethod` decorator.

P.S. Thanks for such a blazingly fast linter and sorter!

--- a/crates/ruff/resources/test/fixtures/pep8_naming/N804.py
+++ b/crates/ruff/resources/test/fixtures/pep8_naming/N804.py
@@ -46,6 +46,10 @@ class MetaClass(ABCMeta):
     def good_method(cls):
         pass
 
+    @staticmethod
+    def static_method(not_cls) -> bool:
+        return False
+
 
 def func(x):
     return x

--- a/crates/ruff/src/ast/function_type.rs
+++ b/crates/ruff/src/ast/function_type.rs
@@ -26,8 +26,18 @@ pub fn classify(
     let ScopeKind::Class(scope) = &scope.kind else {
         return FunctionType::Function;
     };
-    // Special-case class method, like `__new__`.
-    if CLASS_METHODS.contains(&name)
+    if decorator_list.iter().any(|expr| {
+        // The method is decorated with a static method decorator (like
+        // `@staticmethod`).
+        checker.resolve_call_path(expr).map_or(false, |call_path| {
+            staticmethod_decorators
+                .iter()
+                .any(|decorator| call_path == to_call_path(decorator))
+        })
+    }) {
+        FunctionType::StaticMethod
+    } else if CLASS_METHODS.contains(&name)
+        // Special-case class method, like `__new__`.
         || scope.bases.iter().any(|expr| {
             // The class itself extends a known metaclass, so all methods are class methods.
             checker.resolve_call_path(expr).map_or(false, |call_path| {
@@ -46,16 +56,6 @@ pub fn classify(
         })
     {
         FunctionType::ClassMethod
-    } else if decorator_list.iter().any(|expr| {
-        // The method is decorated with a static method decorator (like
-        // `@staticmethod`).
-        checker.resolve_call_path(expr).map_or(false, |call_path| {
-            staticmethod_decorators
-                .iter()
-                .any(|decorator| call_path == to_call_path(decorator))
-        })
-    }) {
-        FunctionType::StaticMethod
     } else {
         // It's an instance method.
         FunctionType::Method

